lib/utils.js

```diff
@@ -45,13 +49,24 @@ async function makeRequest(url, options, parseJson) {
   return res;
 }
 
-async function getLatestUpdate(tld) {
-  const data = await makeRequest(
-    `https://api.hnsnetwork.com/name/records?name=${tld}&limit=50&offset=0`,
-    {},
+const makeRPCRequest = async (method, params = undefined) =>
+  makeRequest(
+    process.env.HSD_RPC_URL,
+    {
+      method: 'POST',
+      auth: `x:${process.env.HSD_API_KEY}`,
+      body: JSON.stringify({
+        method,
+        params,
+      }),
+    },
     true
   );
-  const latestData = data.records.at(0);
+
+async function getLatestUpdate(tld) {
+  const data = await makeRPCRequest('getnameinfo', [tld]);
+  const latestData = data.result.info;
```

Comment:
The only property needed here is the height. And it should be the height of the last UPDATE(/REGISTER). So if a name is being transferred or renewed, then this won't match.